common/app/conf/HealthCheck.scala

```diff
@@ -125,67 +133,65 @@ object HealthCheckPolicy {
+                       (healthChecks: SingleHealthCheck*)
                        (implicit wsClient: WSClient)
   extends HealthCheckController with Results with ExecutionContexts with Logging {
 
-  private[conf] val cache: HealthCheckCache = new HealthCheckCache(wsClient)
+  private[conf] val port: Int = 9000
+
+  private[conf] val cache: HealthCheckCache = new HealthCheckCache(preconditionMaybe)(wsClient)
 
-  private def allSuccessful(results: List[HealthCheckResult]): Boolean = {
-    results match {
-      case Nil => false
-      case nonEmpty => nonEmpty.forall(_.recentlySucceed)
-    }
-  }
-  private def anySuccessful(results: List[HealthCheckResult]): Boolean = results.exists(_.recentlySucceed)
   private def successful(results: List[HealthCheckResult]): Boolean = policy match {
-    case HealthCheckPolicy.All => allSuccessful(results)
-    case HealthCheckPolicy.Any => anySuccessful(results)
+      case HealthCheckPolicy.All => results.nonEmpty && results.forall(_.recentlySucceed)
+      case HealthCheckPolicy.Any => results.exists(_.recentlySucceed)
   }
 
-  val port: Int = 9000
-
   def healthCheck(): Action[AnyContent] = Action.async {
     Future.successful {
       val results = cache.get
-      val response = results.map {
-        r: HealthCheckResult => s"GET ${r.url} '${r.formattedResult}' '${r.formattedDate}'"
-      }
+      val response = results
+        .map { r: HealthCheckResult => s"GET ${r.url} '${r.formattedResult}' '${r.formattedDate}'" }
         .mkString("\n")
+
       if(successful(results)) Ok(response) else ServiceUnavailable(response)
     }
   }
 
-  def runChecks: Future[Unit] = cache.refresh(port, healthChecks:_*).map(_ => Nil)
+  def runChecks(): Future[List[HealthCheckResult]] = cache.refresh(port, healthChecks:_*)
 }
 
 case class AllGoodCachedHealthCheck(healthChecks: SingleHealthCheck*)(implicit wsClient: WSClient)
-  extends CachedHealthCheck(HealthCheckPolicy.All, healthChecks:_*)
+  extends CachedHealthCheck(policy = HealthCheckPolicy.All, preconditionMaybe = None)(healthChecks:_*)
 
 case class AnyGoodCachedHealthCheck(healthChecks: SingleHealthCheck*)(implicit wsClient: WSClient)
-  extends CachedHealthCheck(HealthCheckPolicy.Any, healthChecks:_*)
+  extends CachedHealthCheck(policy = HealthCheckPolicy.Any, preconditionMaybe = None)(healthChecks:_*)
 
 class CachedHealthCheckLifeCycle(
   healthCheckController: CachedHealthCheck,
   jobs: JobScheduler,
-  akkaAsync: AkkaAsync) extends LifecycleComponent {
+  akkaAsync: AkkaAsync) extends LifecycleComponent with ExecutionContexts {
 
   private lazy val healthCheckRequestFrequencyInSec = Configuration.healthcheck.updateIntervalInSecs
 
   override def start() = {
     jobs.deschedule("HealthCheckFetch")
     if (healthCheckRequestFrequencyInSec > 0) {
       jobs.scheduleEvery("HealthCheckFetch", healthCheckRequestFrequencyInSec.seconds) {
-        healthCheckController.runChecks
+        healthCheckController.runChecks().map(_ => ())
```

Comment:
Returning a `Future[Unit]` is a requirement of `LifecycleComponent`, not the healthcheck controller. Therefore it should happen here